services/pull/update.go

```diff
@@ -100,3 +113,83 @@ func GetDiverging(pr *models.PullRequest) (*git.DivergeObject, error) {
 	diff, err := git.GetDivergingCommits(tmpRepo, "base", "tracking")
 	return &diff, err
 }
+
+func doRebase(pr *models.PullRequest, doer *models.User) error {
+	// 1. Clone base repo.
+	tmpBasePath, err := createTemporaryRepo(pr)
+	if err != nil {
+		log.Error("CreateTemporaryPath: %v", err)
+		return err
+	}
+	defer func() {
+		if err := models.RemoveTemporaryPath(tmpBasePath); err != nil {
+			log.Error("Update-By-Rebase: RemoveTemporaryPath: %s", err)
+		}
+	}()
+
+	baseBranch := "base"
+	trackingBranch := "tracking"
+
+	// 2. checkout base branch
+	msg, err := git.NewCommand("checkout", baseBranch).RunInDir(tmpBasePath)
```

Comment:
Argh. No checkouts. They can be absolutely huge, take forever and do strange things.

Especially if you're running Windows.

If we can't do this without checking out we shouldn't do it.

Reply:
We already have code f9r rebase merge pulls, just reuse it

Reply:
> Argh. No checkouts. They can be absolutely huge, take forever and do strange things.
> 
> Especially if you're running Windows.
> 
> If we can't do this without checking out we shouldn't do it.

After carefullly check, I found this step is not needed because it already at this branch. so can rebase directly.
has check it in windows also, and the result is same.

But looks `rebase` will load full work space, I wonder ...

Reply:
> We already have code f9r rebase merge pulls, just reuse it

I hase check it, And if I full understood it. I think they are different.

update pr by rebase:
```
git checkout head-branch 
git rebase base-branch
git push -f head-branch 
```

merge pull request by rebase:
```SHELL
git checkout head-branch 
git rebase base-branch
git checkout base-branch
git merge --ff-only head-branch
git push  base-branch
```